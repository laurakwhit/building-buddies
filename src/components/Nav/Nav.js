import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import './Nav.scss';

const Nav = () => {
  return (
    <div className='nav__section'>
      <img src={require('../../assets/profile.jpg')} alt='profile picture'/>
      <nav>
        <NavLink
          className='nav__section-links'
          activeClassName='selected'
          to='/profile'>
          My Profile
        </NavLink>
        <NavLink
          className='nav__section-links'
          activeClassName='selected'
          to='/neighbors'>
          My Neighbors
        </NavLink>
        <NavLink
          className='nav__section-links'
          activeClassName='selected'
          to='/building'>
          Building Info
        </NavLink>
        <NavLink
          className='nav__section-links'
          activeClassName='selected'
          to='/'>
          Log Out
        </NavLink>
      </nav>
    </div>
  )
}

export default Nav;

Nav.propTypes = {

};
