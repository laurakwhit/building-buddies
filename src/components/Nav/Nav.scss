.nav__section {
  height: 100vh;
  width: 20rem;
  font-size: 2rem;

  img {
    height: 25%;
    margin-top: 3rem;
    border-radius: 50%;
  }
  nav {
    display: flex;
    flex-direction: column;
    height: 55%;
    margin-top: 2rem;

    .nav__section-links {
      padding: 1.2rem 0;
      text-decoration: none;
      color: #112028;

      &:hover {
        color: #fff;
        background: #0f2027; /* fallback for old browsers */
        background: -webkit-linear-gradient(
          to right,
          #2c5364,
          #203a43,
          #0f2027
        ); /* Chrome 10-25, Safari 5.1-6 */
        background: linear-gradient(
          to right,
          #2c5364,
          #203a43,
          #0f2027
        ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
      }
    }
    .selected {
      color: #fff;
      background: #0f2027; /* fallback for old browsers */
      background: -webkit-linear-gradient(
        to right,
        #2c5364,
        #203a43,
        #0f2027
      ); /* Chrome 10-25, Safari 5.1-6 */
      background: linear-gradient(
        to right,
        #2c5364,
        #203a43,
        #0f2027
      ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    }
  }
  background-color: rgb(136, 172, 186);
}
