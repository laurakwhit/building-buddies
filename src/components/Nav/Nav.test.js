import React from 'react';
import { shallow } from 'enzyme';

import Nav from './Nav';

describe('Nav component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav />);
  });

  it('should match the snapshhot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
