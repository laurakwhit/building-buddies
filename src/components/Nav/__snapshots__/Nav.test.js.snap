// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should match the snapshhot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="profile"
          src="profile.jpg"
        />,
        <nav>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            className="nav__section-links"
            to="/profile"
          >
            My Profile
          </NavLink>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            className="nav__section-links"
            to="/neighbors"
          >
            My Neighbors
          </NavLink>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            className="nav__section-links"
            to="/building"
          >
            Building Info
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav__section-links"
            to="/account"
          >
            Account Settings
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav__section-links"
            to="/"
          >
            Log Out
          </NavLink>
        </nav>,
      ],
      "className": "nav__section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "profile",
          "src": "profile.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav__section-links"
              to="/profile"
            >
              My Profile
            </NavLink>,
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav__section-links"
              to="/neighbors"
            >
              My Neighbors
            </NavLink>,
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav__section-links"
              to="/building"
            >
              Building Info
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav__section-links"
              to="/account"
            >
              Account Settings
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav__section-links"
              to="/"
            >
              Log Out
            </NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": "My Profile",
              "className": "nav__section-links",
              "to": "/profile",
            },
            "ref": null,
            "rendered": "My Profile",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": "My Neighbors",
              "className": "nav__section-links",
              "to": "/neighbors",
            },
            "ref": null,
            "rendered": "My Neighbors",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": "Building Info",
              "className": "nav__section-links",
              "to": "/building",
            },
            "ref": null,
            "rendered": "Building Info",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Account Settings",
              "className": "nav__section-links",
              "to": "/account",
            },
            "ref": null,
            "rendered": "Account Settings",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Log Out",
              "className": "nav__section-links",
              "onClick": undefined,
              "to": "/",
            },
            "ref": null,
            "rendered": "Log Out",
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="profile"
            src="profile.jpg"
          />,
          <nav>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav__section-links"
              to="/profile"
            >
              My Profile
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav__section-links"
              to="/neighbors"
            >
              My Neighbors
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav__section-links"
              to="/building"
            >
              Building Info
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav__section-links"
              to="/account"
            >
              Account Settings
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav__section-links"
              to="/"
            >
              Log Out
            </NavLink>
          </nav>,
        ],
        "className": "nav__section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "profile",
            "src": "profile.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav__section-links"
                to="/profile"
              >
                My Profile
              </NavLink>,
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav__section-links"
                to="/neighbors"
              >
                My Neighbors
              </NavLink>,
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav__section-links"
                to="/building"
              >
                Building Info
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav__section-links"
                to="/account"
              >
                Account Settings
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav__section-links"
                to="/"
              >
                Log Out
              </NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": "My Profile",
                "className": "nav__section-links",
                "to": "/profile",
              },
              "ref": null,
              "rendered": "My Profile",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": "My Neighbors",
                "className": "nav__section-links",
                "to": "/neighbors",
              },
              "ref": null,
              "rendered": "My Neighbors",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": "Building Info",
                "className": "nav__section-links",
                "to": "/building",
              },
              "ref": null,
              "rendered": "Building Info",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Account Settings",
                "className": "nav__section-links",
                "to": "/account",
              },
              "ref": null,
              "rendered": "Account Settings",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Log Out",
                "className": "nav__section-links",
                "onClick": undefined,
                "to": "/",
              },
              "ref": null,
              "rendered": "Log Out",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
